src/alphafold_impact/pipelines/data_processing_nih_clinical_trials/nodes.py

```diff
@@ -0,0 +1,147 @@
+            join="outer",
+        )
+
+        logger.info(
+            "Concatenated clinical trials with references from partition %s/%s",
+            idx,
+            n_partitions,
+        )
+
+    return ct_with_refs_combined
+
+
+def _reference_string_to_df(row: pd.Series) -> pd.DataFrame:
+    """
+    Processes a row of the DataFrame, converting the reference string to a list of dictionaries.
+    Returns a DataFrame where each dictionary becomes a row, and each key in the dictionaries becomes a column.
+
+    Args:
+        row (pd.Series): A row of the DataFrame.
+
+    Returns:
+        pd.DataFrame: A DataFrame where each row corresponds to a dictionary in the reference list.
+    """
+    return pd.DataFrame(
+        ast.literal_eval(
+            row["ProtocolSection.ReferencesModule.ReferenceList.Reference"]
+        )
+    ).assign(nct_id=row["ProtocolSection.IdentificationModule.NCTId"])
+
+
+def _extract_doi(citation: str) -> str:
+    """
+    Extracts the DOI from a citation string, ensuring not to include a trailing period.
+
+    Args:
+        citation (str): The citation string from which to extract the DOI.
+
+    Returns:
+        str: The extracted DOI, or an empty string if no DOI is found.
+    """
+    # This pattern looks for 'doi: ' followed by a sequence of characters
+    doi_pattern = r"doi: ([\S]+)"
+    if not (match := re.search(doi_pattern, citation)):
+        return ""
+    doi = match[1]
+    # Remove the trailing period if it's the last character
+    if doi.endswith("."):
+        doi = doi[:-1]
+    return doi
+
+
+def trials_links_to_papers(
+    clinical_trials_with_references: pd.DataFrame,
+) -> pd.DataFrame:
+    """
+    Processes clinical trial data with references to papers, to produce a DataFrame
+    which can be used to link NIH clinical trials to research papers via PubMed ID
+    or DOI.
+
+    This function takes a DataFrame containing clinical trial data with references
+    to papers and performs the following data processing steps:
+    1. Creates new columns for each component of the paper reference.
+    2. Creates new column for the DOI.
+    3. Renames and reorders columns.
+
+    Args:
+        clinical_trials_with_references (pd.DataFrame): A DataFrame containing
+            clinical trial data with references to papers.
+
+    Returns:
+        pd.DataFrame: A processed DataFrame containing the following columns:
+            - 'nct_id': The NCT ID of the clinical trial.
+            - 'ref_pmid': The PMID (PubMed ID) of the paper reference.
+            - 'ref_doi': The DOI (Digital Object Identifier) of the paper reference.
+            - 'ref_citation': The citation of the paper reference.
+            - 'ref_type': The type of the paper reference.
+            - 'ref_retraction_list': The retraction list associated with the paper reference.
+    """
+    return (
+        clinical_trials_with_references.apply(_reference_string_to_df, axis=1)
```

Comment:
FYI this defeats a bit the purpose of pipelines, as this one currently only uses the parent function to process an entire pipeline under the hood that is not quite captured by `kedro viz`. Again, mostly a styling thing at this point, but for future pipelines that may require versioning and better separation of concerns, we may need to revisit pipelines :)